rest/src/main/java/com/redhat/ipaas/rest/v1/dao/ConnectionDAO.java

```diff
@@ -15,26 +15,30 @@
  */
 package com.redhat.ipaas.rest.v1.dao;
 
+import com.redhat.ipaas.rest.v1.controller.handler.rtdb.RealTimeDB;
+import com.redhat.ipaas.rest.v1.model.connection.Connection;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
 
-import java.util.Arrays;
-import java.util.List;
+import javax.inject.Named;
 
-@Component
-public class DefaultDataAccessObjectProvider implements DataAccessObjectProvider {
-
-    private final IntegrationDAO integrationDAO;
-
-    private final IntegrationPatternDAO integrationPatternDAO;
+@Named(Connection.KIND)
+@Service
+public class ConnectionDAO extends RealTimeAbstractDAO<Connection> {
```

Comment:
Another point for view objects ;-) However we _need_ a common language and actually although this is a DB scheme it is directly translatable to the domain object model which is how we handle stuff and how things are related to each other.

The REST service is free to call the several DAOs to construct the appropriate, client friendly response.